### DISCUSSION TITLE
does unified@10 intentionally disallow `Promise<void>` as plugin return type?

### DISCUSSION THREAD

**Discussion by stefanprobst** [unifiedjs/unified#155] [Q&A] [answered]
it looks like the new `unified` version enforces `Promise<undefined>` vs `Promise<void>` as async plugin return type - just wanted to ask if this is intentional?

```js
import {unified} from 'unified'

const processor = unified().use(() => {
  return transformer
  async function transformer() {
    // noop
    return undefined // <-- required
  }
})
```

**Comment by wooorm** [comment, marked as answer]
Good catch. Landed: https://github.com/unifiedjs/unified/commit/591b0c0c24de8d78175c49d15ee43681726e4caf. And released!
